overlord/ifacestate/helpers.go

```diff
@@ -777,10 +777,12 @@ func connectDisconnectAffectedSnaps(t *state.Task) ([]string, error) {
 	return []string{plugRef.Snap, slotRef.Snap}, nil
 }
 
-func ensureSystemSnapIsPresent(st *state.State) error {
+func checkSystemSnapIsPresent(st *state.State) error {
+	// "system" gets remapped to either snapd or a core snap.
+	systemSnap := mapper.RemapSnapFromRequest("system")
```

Comment:
wondering if it's worth introducing mapper.SystemSnapName() to mean this?

Reply:
Will do in a followup.

Reply:
Added with #6146